app/controllers/criteria_controller.rb

```diff
@@ -115,9 +115,15 @@ def update
   def update_positions
     @assignment = Assignment.find(params[:assignment_id])
 
-    ApplicationRecord.transaction do
+    Criterion.transaction(requires_new: true) do
       params[:criterion].each_with_index do |id, index|
-        Criterion.update(id, position: index + 1) unless id.blank?
+        begin
```

Comment:
You don't need a `begin` here. Instead, you can attach the `rescue StandardError` to the end of the transaction `do` block, like here:

https://github.com/MarkUsProject/Markus/blob/master/app/controllers/criteria_controller.rb#L151-L157